msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -2224,6 +2236,220 @@ public String processActionToggleDisplayExtendedDescription()
 
   }
 
+  private void processActionSendToCalendar(MutableEntity forumItem) {  //send the open and close dates of a Forum or Forum Topic to the Calendar. MutableEntity is a parent class of both DiscussionForums and DiscussionTopics.
+    String calendarBeginId;   //these values are used throughout the creation and maintenance of the Calendar sending.
+    String calendarEndId;
+    Boolean sendOpenCloseToCalendar;
+    Boolean availabilityRestrictedNow;
+    String title;
+    Date openDate;
+    Date closeDate;
+    String openingTitle;
+    String closingTitle;
+    Set membershipItems;
+    if (forumItem instanceof DiscussionTopic){  //this part of the processing determines if the MutableEntity is a Forum or a Topic and fills in all the variables by appropriately casting.
+      calendarBeginId = ((DiscussionTopic) forumItem).getCalendarBeginId();
```

Comment:
Downcasting and then calling a method is expensive. I realise it's pre optimisation, but I'd just cast it once and then use that reference. If you're only casting once, then fine, but you're casting many times here.